addons/stock/models/stock_production_lot.py

```diff
@@ -26,7 +29,7 @@ class ProductionLot(models.Model):
     product_qty = fields.Float('Quantity', compute='_product_qty')
     note = fields.Html(string='Description')
     display_complete = fields.Boolean(compute='_compute_display_complete')
-    company_id = fields.Many2one('res.company', 'Company', required=True, stored=True, index=True)
+    company_id = fields.Many2one('res.company', 'Company', required=True, stored=True, index=True, default=_default_company)
```

Comment:
```suggestion
    company_id = fields.Many2one('res.company', 'Company', required=True, stored=True, index=True, default=lambda self: self.env.user.company_id)
```

This could be another solution. @nim-odoo what do you think ?